pkg/database/billing.go

```diff
@@ -29,48 +36,61 @@ type Billing interface {
 		{
-			ID:               "setCreationBillingTimeStamp",
-			TriggerOperation: cosmosdb.TriggerOperationCreate,
-			TriggerType:      cosmosdb.TriggerTypePre,
-			Body: `function trigger() {
-	var request = getContext().getRequest();
-	var body = request.getBody();
-	var date = new Date();
-	var now = Math.floor(date.getTime() / 1000);
-	var billingBody = body["billing"];
-	if (!billingBody["creationTime"]) {
-		billingBody["creationTime"] = now;
-	}
-	request.setBody(body);
-}`,
+			ID: to.Ptr("setCreationBillingTimeStamp"),
+			Body: to.Ptr(`function trigger() {
+				var request = getContext().getRequest();
+				var body = request.getBody();
+				var date = new Date();
+				var now = Math.floor(date.getTime() / 1000);
+				var billingBody = body["billing"];
+				if (!billingBody["creationTime"]) {
+					billingBody["creationTime"] = now;
+				}
+				request.setBody(body);
+			}`),
+			TriggerOperation: to.Ptr(armcosmos.TriggerOperation("Create")),
+			TriggerType:      to.Ptr(armcosmos.TriggerType("Pre")),
 		},
 		{
-			ID:               "setDeletionBillingTimeStamp",
-			TriggerOperation: cosmosdb.TriggerOperationReplace,
-			TriggerType:      cosmosdb.TriggerTypePre,
-			Body: `function trigger() {
-	var request = getContext().getRequest();
-	var body = request.getBody();
-	var date = new Date();
-	var now = Math.floor(date.getTime() / 1000);
-	var billingBody = body["billing"];
-	if (!billingBody["deletionTime"]) {
-		billingBody["deletionTime"] = now;
-	}
-	request.setBody(body);
-}`,
+			ID: to.Ptr("setDeletionBillingTimeStamp"),
+			Body: to.Ptr(`function trigger() {
+				var request = getContext().getRequest();
+				var body = request.getBody();
+				var date = new Date();
+				var now = Math.floor(date.getTime() / 1000);
+				var billingBody = body["billing"];
+				if (!billingBody["creationTime"]) {
+					billingBody["creationTime"] = now;
+				}
+				request.setBody(body);
+			}`),
+			TriggerOperation: to.Ptr(armcosmos.TriggerOperation("Replace")),
+			TriggerType:      to.Ptr(armcosmos.TriggerType("Pre")),
 		},
 	}
 
-	triggerc := cosmosdb.NewTriggerClient(collc, collBilling)
-	for _, trigger := range triggers {
-		_, err := triggerc.Create(ctx, trigger)
-		if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusConflict) {
+	for _, triggerResource := range triggerResources {
+		createUpdateSQLTriggerParameters := armcosmos.SQLTriggerCreateUpdateParameters{
+			Properties: &armcosmos.SQLTriggerCreateUpdateProperties{
+				Options:  &armcosmos.CreateUpdateOptions{},
+				Resource: triggerResource,
+			},
+			Location: &location,
+		}
+		ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+		defer cancel()
+
+		poller, err := sqlResourceClient.BeginCreateUpdateSQLTrigger(ctx, resourceGroup, dbAccountName, dbName, billingContainerName, *triggerResource.ID, createUpdateSQLTriggerParameters, nil)
+		if err != nil {
+			return nil, err
+		}
+		_, err = poller.PollUntilDone(ctx, nil)
+		if err != nil {
```

Comment:
the error logic handling is changing here.  Should it be?  Should we only ignore errors of `http.StatusConflict` from cosmosdb?

Reply:
I ran this multiple times on the same cosmosdb database and it didn't return `http.StatusConflict` even when the triggers were already created. However, it will be good that I just add the logic to ignore `http.StatusConflict` err.